### ISSUE TITLE
Player models are randomly replaced with a default black/blank player model thing.

### ISSUE THREAD

**Issue by XboxFest** [adamdburton/pointshop#178]
Randomly people look like the black blank player model that you see sometimes as the preview of player models in the pointshop menu.
Any ideas?

**Comment by XboxFest** [comment]
Welp I couldn't figure it out and decided to shut down my server... Oh well.

**Comment by nystinicus** [comment]
![2015-01-01_00001](https://cloud.githubusercontent.com/assets/5751862/5593943/5e9b4d2c-91f6-11e4-8bc2-7e1f3cde8395.jpg)

Like this?

**Comment by Paramo96** [comment]
Hope someone finds a fix. My TTT server has been running for over a year and still gets a decent amouny of daily traffic  I don't want this issue to affect it in the long term.

**Comment by Paramo96** [comment]
Amount*

**Comment by iluh** [comment]
I have exactly the same problem. I've just made a local SteamCMD server to play with friends and I added some playermodels to pointshop. They work but they change their colour usually to black / green / blue / dark grey / black flickering, ...

I confirmed this with 'sv_cheats 1' and 'thirdperson' commands. Besides, friends also
![2015-01-09_00001](https://cloud.githubusercontent.com/assets/10457303/5673971/4f775b4e-97a3-11e4-9453-26c561655caa.jpg)
![2015-01-09_00002](https://cloud.githubusercontent.com/assets/10457303/5673972/4f77fe6e-97a3-11e4-942f-b7edc97edbab.jpg)
![2015-01-09_00003](https://cloud.githubusercontent.com/assets/10457303/5673970/4f75d5b2-97a3-11e4-84ca-f242ab34dc90.jpg)

Besides, after preparing round, sometimes the model changes to a default cs terrorist one...
I don't know what else should I do :/

**Comment by Paramo96** [comment]
@iluh  I don't have the problem, only the problem in the screenshot before yours

**Comment by Pugga123** [comment]
I had this problem on my server. Try using TTT Weapon Placer by BadUragrian on Workshop. Make new players spawns. And set ttt_playercolor_mode to 0...

**Comment by Pugga123** [comment]
Also if u are running SteamCMD server u need fast PC and CPU.. Sometimes it happens becaus CPU slow and make mistakes in database...

**Comment by nystinicus** [comment]
@iluh 
That can be fixed with a few simple commands in the config and to a file in the terrortown directory. 
Place this in your server.cfg
ttt_playercolor_mode 0

Go to gamemodes/terrortown/gamemode/player_ext­.lua and edit line 272 by adding -- or // at the beginning

Your problem is not the same as ours. :/

**Comment by iluh** [comment]
Thanks to all of you! Done! :)

**Comment by Paramo96** [comment]
umm does that fix the other proble in the first screenshot? because these apear to be 2 different things....

**Comment by nystinicus** [comment]
No it fixes the stuff in the second screenshot. We are still having the problem. Also, i think you guys should update your ttt, Im having less problems with it now.

**Comment by Paramo96** [comment]
update ttt, how lol. Thought it updated auto. less problems, how?

**Comment by nystinicus** [comment]
Now im having like 0 errors but you get it from here https://github.com/garrynewman/garrysmod

**Comment by Paramo96** [comment]
Not sure if I'm able to update TTT since my server is hosted by NFO (managed?)

**Comment by cygiel** [comment]
I know this thread is old, but I recently run into the same issue with playermodels turning into blank playermodels. Well, I fixed it. 

"Sometimes they have it at the begining and later on in the round its back to normal. Seems to be random. Also if you kill a player with the blank model their model goes back to normal."
Same thing happened to me...


"Also if u are running SteamCMD server u need fast PC and CPU.. Sometimes it happens becaus CPU slow and make mistakes in database..."
Kind of, not exactly. You might have a powerful Server with the best CPU, but if you have a lot of addons, then they obviously use resources and, as mentioned in that quote, it can lead to errors in database. That's what really happens. I'd also suggest updating your server.cfg with lag compensation convars and other tweaks to improve gaming experience. That's what I did as well.


"Go to gamemodes/terrortown/gamemode/player_ext­.lua and edit line 272 by adding -- or // at the beginning"
No. Definately don't do that. Ever. If you have a problem as described above, that playermodels change to blank playermodels, that clearly means that they're working but the game is checking the code for playermodels too early when starting a round and it's loading that piece of code alongside all of the addons, creating errors in database (etc etc), so to fix that, follow this little guideline. 100% guaranteed it'll work. I tested it myself...

Fix:
Just copy "hook.Call("PlayerSetModel", GAMEMODE, self)" to appropriate places in the player_ext.lua file, like so...

Starting from line 269...

-- Preps a player for a new round, spawning them if they should. If dead_only is
-- true, only spawns if player is dead, else just makes sure he is healed.
function plymeta:SpawnForRound(dead_only)
   hook.Call("PlayerSetModel", GAMEMODE, self)
   hook.Call("TTTPlayerSetColor", GAMEMODE, self)

   -- wrong alive status and not a willing spec who unforced after prep started
   -- (and will therefore be "alive")
   if dead_only and self:Alive() and (not self:IsSpec()) then
      -- if the player does not need respawn, make sure he has full health
      self:SetHealth(self:GetMaxHealth())
	  hook.Call("PlayerSetModel", GAMEMODE, self)
      return false
   end

   if not self:ShouldSpawn() then return false end

   -- reset propspec state that they may have gotten during prep
   PROPSPEC.Clear(self)

   -- respawn anyone else
   if self:Team() == TEAM_SPEC then
      self:UnSpectate()
	  hook.Call("PlayerSetModel", GAMEMODE, self)
   end

   self:StripAll()
   self:SetTeam(TEAM_TERROR)
   self:Spawn()

   -- tell caller that we spawned
   return true
end

function plymeta:InitialSpawn()
   self.has_spawned = false

   -- The team the player spawns on depends on the round state
   self:SetTeam(GetRoundState() == ROUND_PREP and TEAM_TERROR or TEAM_SPEC)

   -- Change some gmod defaults
   self:SetCanZoom(false)
   self:SetJumpPower(160)
   self:SetSpeed(false)
   self:SetCrouchedWalkSpeed(0.3)

   -- Always spawn innocent initially, traitor will be selected later
   self:ResetStatus()

   -- Start off with clean, full karma (unless it can and should be loaded)
   self:InitKarma()

   -- We never have weapons here, but this inits our equipment state
   self:StripAll()
   
   hook.Call("PlayerSetModel", GAMEMODE, self)
end

Ending at line 325.

Simple, effective, awesome, clever.

:)

Btw, to fix this issue, it took me less than 3 minutes. Most of the time I tried to figure out what exactly was happening, then I googled the issue and found this thread :)
